src/components/Shared/FormShell.js

```diff
@@ -357,6 +358,19 @@ export default function FormShell({
                 })
               }
               break
+            case 'onItemPromotion':
+              action.onClick = () => {
+                stack({
+                  Component: ItemPromotion,
+                  props: {
+                    invoiceId: action.invoiceId
+                  },
+                  width: 1330,
+                  height: 720,
+                  title: platformLabels.ItemPromotion
+                })
+              }
+              break
```

Comment:
is this is a common button? if not please put it in the page itself, if we are using it in more that 3 pages leave it